lib/chai/interface/assert.js

```diff
@@ -590,6 +591,56 @@ assert.isNotFunction = function (val, msg) {
   new Assertion(val, msg, assert.isNotFunction, true).to.not.be.a('function');
 };
 
+/**
+ * TODO
+ */
+assert.isCallable = function (val, msg) {
+  if (!['Function', 'AsyncFunction', 'GeneratorFunction', 'AsyncGeneratorFunction'].includes(type(val))) {
+    throw new AssertionError(
+      msg + ': expected ' + inspect(val) + ' to be callable',
+      undefined,
+      assert.isCallable
+    );
+  }
+}
+
+/**
+ * ### .isAsyncFunction(value, [message])
+ *
+ * Asserts that `value` is a async function.
```

Comment:
I am not a native english speaker

```suggestion
 * Asserts that `value` is an async function.
```

Reply:
Heh, yeah me neither and I always get a/an mixed up. Nice catch :)